lib/AST/LookupVisibleDecls.cpp

```diff
@@ -551,13 +602,13 @@ static void lookupVisibleMemberDeclsImpl(
   }
 
   // Enumerate members of archetype's requirements.
-  if (ArchetypeType *Archetype = BaseTy->getAs<ArchetypeType>()) {
-    for (auto Proto : Archetype->getConformsTo())
-      lookupVisibleProtocolMemberDecls(
-          BaseTy, Proto->getDeclaredType(), Consumer, CurrDC, LS,
-          getReasonForSuper(Reason), TypeResolver, GSB, Visited);
-
-    if (auto superclass = Archetype->getSuperclass())
+  if (ArchetypeType *AT = BaseTy->getAs<ArchetypeType>()) {
```

Comment:
Is this a non-functional change? Can you split it off into a separate patch if it’s unrelated?